### PR TITLE
Deprecate service nomenclature

### LINKED ISSUE #776: support namespace:kind/name syntax in fluxctl
fluxctl should support the same `namespace:kind/name` syntax as is used in the UI, so that's easier for users to switch between and c&p between CLI and UI.

### PR DESCRIPTION
**Pull request by awh**
### Mechanical gorenames of types and constants

See the commit message for a summary of the renames. You will note that in some cases 'Service' is replaced by 'Resource' and in other cases by 'Controller' - this split is because sometimes (starting with `flux.ResourceID` itself) these types can be used to refer to _any_ kind of cluster resource, but in other cases (such as `flux.ControllerStatus`) they obviously refer specifically to pod controllers only. 

### Changes to the `fluxctl` UI

* The `list-services` command is no longer supported, and now returns a deprecation error (not a warning!) advising the use of `list-controllers` instead
* Similarly, commands previously accepting `--service` will now return a deprecation error if this flag is passed and advise the use of `--controller` instead

The rationale for this decision (rather than a deprecation warning) is that the 'new' fluxd can't provide meaningful behaviour WRT the old `<namespace>/<servicename>` identifier (other than perhaps guessing that there is also a deployment in `<namespace>` called `<servicename>`, and using that) so it seemed safest to generate a hard error.

Fixes #776.

### CODE REVIEW

**Comment by squaremo** [comment]
The rationale is :100: I reckon.

File: remote/errors.go [hunk 1]
```diff
@@ -56,14 +56,13 @@ func UnsupportedResourceKind(err error) error {
...
-Important - releasing by service name is no longer supported - if
-you're using an old version of fluxctl that does not let you specify
-a kind other than 'service', you will need to upgrade it.
+The version of the agent running in your cluster (fluxd) can only release
+updates to pod controllers such as Deployments, DaemonSets, StatefulSets and
+CronJobs.
+
+Releasing by service name is no longer supported - if you're using an old
+version of fluxctl that does not let you specify a kind other than 'service',
+you will need to upgrade it.
```
Comment (squaremo) [inline remote/errors.go]: Yeah, nice.

File: cmd/fluxctl/list_services_cmd.go [hunk 2]
```diff
@@ -23,71 +17,15 @@ func newServiceList(parent *rootOpts) *serviceListOpts {
...
 func (opts *serviceListOpts) Command() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "list-services",
-		Short:   "List services currently running on the platform.",
-		Example: makeExample("fluxctl list-services"),
-		RunE:    opts.RunE,
+		Use:    "list-services",
+		Short:  "Deprecated - use list-controllers instead",
+		Hidden: true,
+		RunE:   opts.RunE,
```
Comment (squaremo) [inline cmd/fluxctl/list_services_cmd.go]: :white_check_mark:

File: cmd/fluxctl/lock_cmd.go [hunk 3]
```diff
@@ -6,37 +6,52 @@ import (
...
 		RunE: opts.RunE,
 	}
 	AddOutputFlags(cmd, &opts.outputOpts)
 	AddCauseFlags(cmd, &opts.cause)
+	cmd.Flags().StringVarP(&opts.namespace, "namespace", "n", "default", "Controller namespace")
+	cmd.Flags().StringVarP(&opts.controller, "controller", "c", "", "Controller to lock")
+
+	// Deprecated
 	cmd.Flags().StringVarP(&opts.service, "service", "s", "", "Service to lock")
+	cmd.Flags().MarkHidden("service")
```
Comment (squaremo) [inline cmd/fluxctl/lock_cmd.go]: Isn't there a `MarkDeprecated`?

Comment (awh) [reply cmd/fluxctl/lock_cmd.go]: There is, but it just prints a warning message and then continues as normal. This meant I had to manually test to see if the arg was supplied and then error out, so I just used `MarkHidden` + the normal error return path to deliver the message.

Comment (squaremo) [reply cmd/fluxctl/lock_cmd.go]: Got it :+1:

File: remote/errors.go (hunk 1 above)
Comment (squaremo) [reply remote/errors.go]: It occurs to me that this could be generated from the value in resourcekinds.go. Gilding the lily perhaps ...

**Pull request merged**